adapters/sinatra/lib/opentelemetry/adapters/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      module Middlewares
+        class TracerMiddleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            tracer.in_span(
+              env['PATH_INFO'],
+              attributes: { 'component': 'http',
+                            'http.method': env['REQUEST_METHOD'] },
+              kind: :server,
+              with_parent_context: parent_context(env)
+            ) do |span|
+              trace_request(span, env)
+
+              app.call(env).tap { |resp| trace_response(span, env, resp) }
+            end
+          end
+
+          private
+
+          attr_reader :app
+
+          def parent_context(env)
+            OpenTelemetry.tracer_factory.http_text_format.extract(env)
+          end
+
+          def tracer
+            OpenTelemetry::Adapters::Sinatra::Adapter.tracer
+          end
+
+          def trace_request(span, env)
+            span.set_attribute('http.url', env['PATH_INFO'])
+          end
+
+          def trace_response(span, env, resp)
+            status, _headers, _response_body = resp
+
+            span.set_attribute('http.status_code', status)
```

Comment:
We should set the `span.status` here as well.

Reply:
Hopefully on the right track here... I had to add `Trace::Status.from_http_status(code) => Status`, which didn't _exactly_ map 1-to-1 http -> gRPC status.